packages/tupaia-web/src/features/Visuals/View/DownloadFiles.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+import { darken } from '@material-ui/core';
+import { useSearchParams } from 'react-router-dom';
+import { ViewConfig } from '@tupaia/types';
+import { DownloadFilesVisual } from '@tupaia/ui-components';
+import { useDownloadFiles } from '../../../api/mutations';
+import { URL_SEARCH_PARAMS } from '../../../constants';
+import { ViewReport } from '../../../types';
+
+const StyledDownloadFilesVisual = styled(DownloadFilesVisual)`
+  .filename {
+    color: ${({ theme }) => darken(theme.palette.common.white, 0.1)};
+  }
+  .checkbox-icon {
+    color: ${({ theme }) => darken(theme.palette.common.white, 0.1)};
+  }
+`;
+
+interface DownloadFilesVisualProps {
+  report: ViewReport;
+  config: ViewConfig;
+  isEnlarged?: boolean;
+}
+
+export const DownloadFiles = ({
+  report: { data = [] },
+  config,
+  isEnlarged,
+}: DownloadFilesVisualProps) => {
+  const [urlSearchParams, setUrlSearchParams] = useSearchParams();
+
+  const { mutate: download, error, reset } = useDownloadFiles();
+
+  const onClose = () => {
+    reset();
+    urlSearchParams.delete(URL_SEARCH_PARAMS.REPORT);
+    setUrlSearchParams(urlSearchParams);
+  };
+
+  const errorObj = error as Error;
+
+  return (
+    <StyledDownloadFilesVisual
+      config={config}
+      data={(data as unknown) as { uniqueFileName: string; label: string }[]}
+      downloadFiles={download as (uniqueFileNames: string[]) => Promise<void>}
```

Comment:
The shape of the data for the download files viz is different to all the other Views so this seems like the best way to handle the type casting to me